src/Teams/Data/Repositories/MultipleAnswerQuestionRepository.cs

```diff
@@ -9,14 +9,31 @@ namespace Teams.Data.Repositories
 {
     public class MultipleAnswerQuestionRepository : IMultipleAnswerQuestionRepository
     {
-        ApplicationDbContext _db;
-        public MultipleAnswerQuestionRepository(ApplicationDbContext db)
+        IApplicationDbContext _db;
+        public MultipleAnswerQuestionRepository(IApplicationDbContext db)
         {
             _db = db;
         }
         public MultipleAnswerQuestion PickById(Guid Id)
         {
             return _db.MultipleAnswerQuestions.Include(a => a.Answers).Single(q => q.Id == Id);
         }
+
+        public void AddQuestion(MultipleAnswerQuestion question)
+        {
+            _db.MultipleAnswerQuestions.Add(question);
+        }
+
+        public void DeleteQuestionOptionsIn_DB(MultipleAnswerQuestion question)
+        {
+            question.Answers.ToList().ForEach(o => _db.Entry(o).State = EntityState.Deleted);
```

Comment:
Converting ReadOnlyCollection to List each time just to access the ForEach extension is not a good idea.
Just iterate over the collection with default foreach(), or create a custom ForEach extension for ReadOnlyCollection type.